pallets/cdd-offchain-worker/src/lib.rs

```diff
@@ -216,49 +253,40 @@ impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
     /// are being exempted and marked as valid.
     fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
         // Firstly let's check that we call the right function.
-        if let Call::take_off_invalidate_nominators(block_number, target, signature) = call {
+        if let Call::submit_unsigned_invalidate_nominators_with_signed_payload(
+            ref payload,
+            ref signature,
+        ) = call
+        {
             // Now let's check if the transaction has any chance to succeed.
             let last_unsigned_at = <LastExtSubmittedAt<T>>::get();
             ensure!(
-                last_unsigned_at + T::CoolingInterval::get() <= *block_number,
+                last_unsigned_at + T::CoolingInterval::get() <= payload.block_number,
                 InvalidTransaction::Stale
             );
 
             // Let's make sure to reject transactions from the future.
             ensure!(
-                &<system::Module<T>>::block_number() >= block_number,
+                &<system::Module<T>>::block_number() >= &payload.block_number,
                 InvalidTransaction::Future
             );
 
-            // Get the key from the store to verify whether the signed signature is
-            // generated from the validator or not.
-            let signer_id = T::SignerId::all()
-                .into_iter()
-                .next()
-                .ok_or(InvalidTransaction::BadProof)?;
-
-            let signature_valid = target.using_encoded(|encoded_nominators| {
-                signer_id.verify(&encoded_nominators, &signature)
-            });
             // validating signature
-            ensure!(signature_valid, InvalidTransaction::BadProof);
-
-            Ok(ValidTransaction {
-                // We set the priority to the max value - 10000. ~ near to high priority
-                // TODO: We can change after discussion
-                priority: TransactionPriority::max_value() - 10000_u64,
-                requires: vec![],
-                // We set the `provides` tag to be the same as `last_unsigned_at`. This makes
-                // sure only one transaction produced after `last_unsigned_at` will ever
-                // get to the transaction pool and will end up in the block.
-                // We can still have multiple transactions compete for the same "spot",
-                // and the one with higher priority will replace other one in the pool.
-                provides: vec![codec::Encode::encode(&(KEY_TYPE.0, last_unsigned_at))],
-                // The transaction is only valid for next n blocks. After that it's
-                // going to be revalidated by the pool. n = cooling interval
-                longevity: (T::CoolingInterval::get()).saturated_into::<u64>(),
-                propagate: true,
-            })
+            ensure!(
+                SignedPayload::<T>::verify::<T::AuthorityId>(payload, signature.clone()),
```

Comment:
I think this will only check the sign against local keys. If we want the tx to remain local, we don't need to sign it. We can simply check if the tx originated locally using the same approach that phragmen uses. https://github.com/PolymathNetwork/Polymesh/blob/7837422c498f682dfed70eb989ab8abe36a57ca7/pallets/staking/src/lib.rs#L3997